src/library.py

```diff
@@ -910,82 +912,259 @@ def verify(self):
             ):  # remove the Jan 1st of next year reason: the pandas date for loop iterates one more loop is hour is 24:00:00
                 pass
             else:
-                if (
-                    day["O_sch"] <= tol_occ
-                ).all():  # confirmed this room is NOT rented out
-                    if (day["T_z_hea_set"] < 15.6 + tol_temp).all() and (
-                        day["T_z_coo_set"] > 26.7 - tol_temp
-                    ).all():
-                        result_repo.append(
-                            1
-                        )  # pass, confirmed zone temperature setpoint reset during the unrented period
+                df_s_AHU_diff = day[day["s_AHU"].diff() != 0.0]["s_AHU"]
+                if len(df_s_AHU_diff) > 1:
+                    # unocc to occ
+                    df_occ_diff = day[day["occ"].diff() != 0.0]["occ"]
+
+                    if (
+                        len(df_occ_diff) > 1
+                    ):  # Since first value of .diff() is nan, this means when values don't change
+                        t_unocc_to_occ = df_occ_diff.index.tolist()[1]
                     else:
-                        result_repo.append(
-                            0
-                        )  # fail, zone temperature setpoint was not reset correctly
-                else:  # room is rented out
-                    T_z_hea_occ_set = day.query("O_sch > 0.0")["T_z_hea_set"].max()
-                    T_z_coo_occ_set = day.query("O_sch > 0.0")["T_z_coo_set"].min()
+                        t_unocc_to_occ = day.index[0]
+
+                    t_s_AHU_off_to_on = df_s_AHU_diff.index.tolist()[1]
 
                     if (
-                        day["T_z_hea_set"] < T_z_hea_occ_set - 2.22 + tol_temp
-                    ).all() or (
-                        day["T_z_coo_set"] > T_z_coo_occ_set + 2.22 - tol_temp
-                    ).all():
-                        result_repo.append(
-                            1
-                        )  # pass, confirm the HVAC setpoint control resets when guest room reset when occupants leave the room
+                        pd.Timedelta(t_s_AHU_off_to_on - t_unocc_to_occ).seconds / 3600
```

Comment:
I am not sure if both positive and negative legit? why are we checking for 12 hours.

Reply:
This is because for example, when t_s_AHU_off_to_on: 8:00 am, t_unocc_to_occ:  10:00 am, if I do (t_s_AHU_off_to_on - t_unocc_to_occ), the time difference is 22 hrs instead of 2 hrs. So, I subtract the earlier time stamp from the later time stamp (e.g., in this example, t_unocc_to_occ - t_s_AHU_off_to_on) to make sure the time difference is within 24 hr time frame.

Reply:
So, I just tested with ```delta```, ```value```, and ```seconds```.  And, when I printed these out to see what ```delta``` and ```value``` methods result in, the following is one example
```delta```: ```21600000000000```, ```value```: ```21600000000000```, ```second```:```21600```.
In this case, using either one of them seems fine and I don't see benefits in using ```delta``` and ```value``` because these two values don't inform us more information. How do you think Jerry?

Reply:
when value and delta are negative, second is still positive.